controller/ali-cluster-config-handler.go

```diff
@@ -2,37 +2,288 @@ package controller
 
 import (
 	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
 
+	cs "github.com/alibabacloud-go/cs-20151215/v5/client"
+
+	"github.com/rancher/ali-operator/pkg/alibaba"
+	"github.com/rancher/ali-operator/pkg/alibaba/services"
 	aliv1 "github.com/rancher/ali-operator/pkg/apis/ali.cattle.io/v1"
 	alicontrollers "github.com/rancher/ali-operator/pkg/generated/controllers/ali.cattle.io/v1"
 	wranglerv1 "github.com/rancher/wrangler/v3/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
 )
 
 const (
-	controllerName       = "ali-controller"
-	controllerRemoveName = "ali-controller-remove"
+	aliClusterConfigKind     = "AliClusterConfig"
+	aliConfigCreatingPhase   = "creating"
+	aliConfigNotCreatedPhase = ""
```

Comment:
IMO I don't know if we need a sentinel value for an empty string. When reviewing I thought this was going to be an actual field (jumped around a bit during my review), which makes me feel like it's not super straightforward at first glance.